plugin/storage/memory/memory.go

```diff
@@ -107,20 +107,12 @@ func (st *Store) GetDependencies(ctx context.Context, endTs time.Time, lookback
 		if traceIsBetweenStartAndEnd(startTs, endTs, trace) {
 			for _, s := range trace.Spans {
 				parentSpan := findSpan(trace, s.ParentSpanID())
-				if parentSpan != nil {
-					if parentSpan.Process.ServiceName == s.Process.ServiceName {
-						continue
-					}
-					depKey := parentSpan.Process.ServiceName + "&&&" + s.Process.ServiceName
-					if _, ok := deps[depKey]; !ok {
-						deps[depKey] = &model.DependencyLink{
-							Parent:    parentSpan.Process.ServiceName,
-							Child:     s.Process.ServiceName,
-							CallCount: 1,
-						}
-					} else {
-						deps[depKey].CallCount++
-					}
+				if parentSpan != nil && parentSpan.Process.ServiceName != s.Process.ServiceName {
+					updateServiceDependencyLinks(parentSpan.Process.ServiceName, s.Process.ServiceName, deps)
+				}
+				if isLeaf(s, trace.Spans) {
```

Comment:
A span being a leaf in the call graph does not mean there's some uninstrumented service below it, it could be a genuine leaf. In addition to leaf you also need to check that the span kind is client or producer, as that would definitively indicate that there is supposed to be "the other side" of that transmission, which is not found in the trace

Reply:
Agreed! Being a leaf doesn't necessarily mean an uninstrumented service. I have updated the function.